hbase-server/src/main/java/org/apache/hadoop/hbase/mob/DefaultMobStoreCompactor.java

```diff
@@ -331,6 +327,7 @@ protected boolean performCompaction(FileDetails fd, InternalScanner scanner, Cel
           "Interrupted while control throughput of compacting " + compactionName);
     } finally {
       throughputController.finish(compactionName);
+      ((ShipperListener) writer).beforeShipped();
```

Comment:
Below in Compactor, call beforeShipped() before finishing the throughputController. Its ok for any order. Still we can maintain one order.

Reply:
Yeah,  I will provide a UT to address this bug but am working on other things now... will update this patch later.